### ISSUE TITLE
Reader: anchor "page jump" links don't work in the Reader view

### ISSUE THREAD

**Issue by SiobhyB** [wordpress-mobile/WordPress-Android#7654]
### Expected behavior

When viewing a post that contains anchor links ("page jumps"), I expected to be taken to section that's linked to in the post.

### Actual behavior

Tapping on the the anchor links has no effect when viewed within the app. The links work correctly when selected in a browser (both from the Reader and the site itself).

### Steps to reproduce the behavior

1. Follow the steps in [this support documentation](https://en.support.wordpress.com/splitting-content/page-jumps/) to add "page jumps" to a post.

2. View the post from within the app's Reader and tap on the anchor links.

##### Tested on Pixel, Android 8.1.0, alpha-107

Reported in 275749-h. iOS version of issue: https://github.com/wordpress-mobile/WordPress-iOS/issues/9140.

**Comment by planarvoid** [comment]
The main issue behind this problem is having a `WebView` inside a `ScrollView`. In this case the `WebView` is not handling scrolling and it's not propagating any scrolling to the encapsulating `ScrollView`. There doesn't seem to be an easy fix for this issue. We still need the `ScrollView` to wrap the header and related posts.

**Comment by designsimply** [comment]
Tested and confirmed this is still an issue in WPAndroid alpha-173 (12.6 alpha) on Pixel 3 Android 9.

<img src="https://user-images.githubusercontent.com/1119271/59475170-601ded00-8e07-11e9-852d-ef96e51cf28e.png" width="270" alt="Screenshot_20190613-181217">
<sub>Sample post: https://artindesignco.blog/2019/06/14/page-jump-test/</sub>

**Comment by sarahfg** [comment]
User reached out to us about this #2744961-zen

**Comment by renanferrari** [comment]
After some investigation, I have come to the conclusion that what @planarvoid stated above still holds: the main issue here is that we need to wrap our `WebView` with a `ScrollView`, so we can display the custom views at the top (header) and at the bottom (related posts) of the content, which causes the `WebView` to not handle scroll events. 

As an experiment to confirm that this was indeed the issue, I went as far as forcefully removing the `ScrollView` from that layout just to see if the anchor links would start working and they did. Of course, doing so breaks the rest of the layout.

Here's a list of ideas and quick hacks I tried:
- [Measuring the `WebView` height and manually set its layout params](https://stackoverflow.com/a/34872043/518179)
- Replacing the `ScrollView` with a `NestedScrollView` and set `android:fillViewport="true"`.
- Messing around with some scrolling attributes (like [`scrollContainer`](https://stackoverflow.com/a/5274400/518179)) on both the `ScrollView` and `WebView` to see if there was any way to better handle the scrolling events.
- Replacing the `ScrollView` with a `RecyclerView` (with separate header, content and footer items). In this case anchor links worked, but everything else was quite broken.
- Intercepting the link and [programmatically scroll the `WebView`](https://stackoverflow.com/q/2238938/518179). This also doesn't work, as the `WebView` [doesn't let us handle anchor links](https://developer.android.com/guide/webapps/migrating.html#URLs) with our current configuration. Even if we changed our configuration to allow that, the `WebView` scroll methods also wouldn't work when inside a `ScrollView`, naturally.

The only obvious solution I see at the moment would be to make the use of a `ScrollView` unnecessary by including both header and footer in the post HTML, but that doesn't seem feasible or desirable at the moment.

**Comment by khaykov** [comment]
I also took a look at this issue, and while I got some progress, I didn't have enough time to bring it to a workable state. Here is what I did, and maybe it will come in handy for someone else:

Idea was to intercept clicks on anchor links and use Javascript to get the coordinate of the anchor. This worked well. I haven't tried, but In theory, we could have a JS interface that will return those coordinates to our java/kotlin code.
Using the coordinates returned by JS from the web view, we can scroll the ScrollView to the correct position. 
I don't have much experience with ScrollView, so I wasn't able to finish this, but we need to find the topmost Y coordinate of WebView inside ScrollView (or height of the content above WebView) and add it to the Y coordinate of the anchor that we received from WebView using JS. Adding those two together should tell us where to scroll the ScrollView to show anchor.

There might be some issue with this approach there, of course :)
